logbook-core/src/main/java/org/zalando/logbook/XmlCompactingBodyFilter.java

```diff
@@ -0,0 +1,100 @@
+package org.zalando.logbook;
+
+import lombok.extern.slf4j.Slf4j;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathFactory;
+import java.io.ByteArrayInputStream;
+import java.io.StringWriter;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static javax.xml.transform.OutputKeys.INDENT;
+import static javax.xml.transform.OutputKeys.OMIT_XML_DECLARATION;
+import static javax.xml.xpath.XPathConstants.NODESET;
+import static org.zalando.fauxpas.FauxPas.throwingSupplier;
+
+@Slf4j
+class XmlCompactingBodyFilter implements BodyFilter {
+
+    private final Predicate<String> contentTypes = MediaTypeQuery.compile("*/xml", "*/*+xml");
+    private final DocumentBuilderFactory documentBuilderFactory = documentBuilderFactory();
+    private final Transformer transformer = transformer();
+
+    @Override
+    public String filter(@Nullable final String contentType, final String body) {
+        return contentTypes.test(contentType) && shouldCompact(body) ? compact(body) : body;
+    }
+
+    private boolean shouldCompact(final String body) {
+        return Stream.of("<?xml", "\n", "  ").anyMatch(body::contains);
```

Comment:
The xml declaration is redundant and should be covered by the content type check, shouldn't it? Spaces would be generally ok, since the whole log message would still fit into a single line.

I'd just change this in the same way as the json compactor:

```
return body.indexOf('\n') != -1;
```

Reply:
It's quite possible to have `xml prolog` inside of the message, but do not have `\n` inside.
So, to not check `<?xml` potentially means to keep it in compacted version (due to compact skip)

Reply:
But if there are no newlines, what exactly would you expect to happen when compacting? Removing other whitespaces?

Reply:
Well I got your point. We have different points of view to minimisation. I consider extra whitespaces and xml prolog as too verbose in logs. But I could leave with, cause single line is much more important